src/paste.rs

```diff
@@ -261,10 +271,19 @@ impl Paste {
                 file_name = random_text;
             }
         }
-        let mut path = self
-            .type_
-            .get_path(&config.server.upload_path)
-            .join(&file_name);
+        let mut path = util::safe_path_join(
+            self.type_
+                .get_path(&config.server.upload_path)
+                .ok_or(IoError::new(
+                    IoErrorKind::Other,
+                    String::from("invalid filename"),
+                ))?,
```

Comment:
I'm thinking if it makes more sense to return `Result<PathBuf>` from `safe_path_join` so that we don't need to map the `Option` everytime we use it. What do you think?

Reply:
We could, and I guess create our own `IoError` type to handle it. It ought to make the code a bit clearer - I'll give it a try.